src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+            {
+                window = (JSObject)Interop.Runtime.GetGlobalObject("window");
+                fetch = (JSObject)Interop.Runtime.GetGlobalObject("fetch");
+            }
+
+            public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                // There is a race condition on Safari as a result of using TaskCompletionSource that
+                // causes a stack exceeded error being thrown.  More information can be found here:
+                // https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/
+                var tcs = new TaskCompletionSource<HttpResponseMessage>(TaskCreationOptions.RunContinuationsAsynchronously);
+                using (cancellationToken.Register(() => tcs.TrySetCanceled()))
+                {
+#pragma warning disable 4014
+                    doFetch(tcs, request, cancellationToken).ConfigureAwait(false);
+#pragma warning restore 4014
+
+                    return await tcs.Task.ConfigureAwait(true);
+                }
+            }
+
+            private async Task doFetch(TaskCompletionSource<HttpResponseMessage> tcs, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                if (request == null)
+                {
+                    throw new ArgumentNullException(nameof(request));
+                }
+
+                try
+                {
+                    var requestObject = new JSObject();
+
+                    if (request.Properties.TryGetValue("WebAssemblyFetchOptions", out var fetchOoptionsValue) &&
+                        fetchOoptionsValue is IDictionary<string, object> fetchOptions)
+                    {
+                        foreach (var item in fetchOptions)
+                        {
+                            requestObject.SetObjectProperty(item.Key, item.Value);
+                        }
+                    }
+
+                    requestObject.SetObjectProperty("method", request.Method.Method);
+
+                    // We need to check for body content
+                    if (request.Content != null)
+                    {
+                        if (request.Content is StringContent)
+                        {
+                            requestObject.SetObjectProperty("body", await request.Content.ReadAsStringAsync().ConfigureAwait(true));
+                        }
+                        else
+                        {
+                            // 2.1.801 seems to have a problem with the line
+                            // using (var uint8Buffer = Uint8Array.From(await request.Content.ReadAsByteArrayAsync ()))
+                            // so we split it up into two lines.
+                            var byteAsync = await request.Content.ReadAsByteArrayAsync().ConfigureAwait(true);
+                            using (var uint8Buffer = Uint8Array.From(byteAsync))
```

Comment:
For all of these vars, please replace them with the actual type, except when the type is obvious from a `new` or explicit cast on the right-hand side.

Reply:
Addressed in recent commits